generated/nidcpower/session.py

```diff
@@ -8,6 +8,41 @@
 from nidcpower import library_singleton
 from nidcpower import visatype
 
+import pprint
+pp = pprint.PrettyPrinter(indent=4)
+
+
+class _TimedeltaConverter(object):
+    def __init__(self, value, library_type, library_units):
+        self._value = value
+        self._library_type = library_type
+        self._library_units = library_units
+
+        if str(type(value)).find("'datetime.timedelta'") != -1:
+            if library_units == 'seconds':
+                scaling = 1
+            elif library_units == 'milliseconds':
+                scaling = 1000
+            elif library_units == 'microseconds':
```

Comment:
If we create a type for each of these units (seconds, milliseconds, microseconds) the string comparison goes away. I like that.

Reply:
Done, but not as type but as separate functions.